pkg/local/sync.go

```diff
@@ -305,10 +305,17 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 		ClientMtimeMetadataKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
 	}
 
-	reader := b.Reader(f)
+	readerWrapper := fileWrapper{
+		file:   f,
+		reader: b.Reader(f),
```

Comment:
Why do we need both?

Reply:
ProgressUpdater requires a reader for the progress bar but we also need the seek functionality for the upload so we need to wrap the progress reader with a seek implementation as well